flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/enumerator/cursor/StopCursor.java

```diff
@@ -71,7 +71,7 @@ static StopCursor afterMessageId(MessageId messageId) {
         return new MessageIdStopCursor(messageId, false);
     }
 
-    static StopCursor atEventTime(long timestamp) {
+    static StopCursor atPublishTime(long timestamp) {
```

Comment:
I am afraid this change it not easily doable. The class is annotated with `@PublicEvolving` and this may break existing users setup.

The only way I see is that you can deprecate `atEventTime` and create another method `atPublishTime`. In the next release we can then safely remove `atEventTime`.

Reply:
After some thought, I think maybe we don't really need to deprecated atEventTime. What we can do is to add a publishTime stop cursor and let users choose whichever they want to use.

Reply:
I searched and found something: "All method declarations in an interface, including static methods, are implicitly public, so you can omit the public modifier.  " 
 https://docs.oracle.com/javase/tutorial/java/IandI/defaultmethods.html#static
 
So here users can use these static members.

On the other hand, in general we recommend atPublishTime. I'll delete the documentation for atEventTime().

Reply:
One thing left is that, pyflink has Pulsar connector support, but I don't know if that file is auto-generated or maintained by developers. 

flink-python/pyflink/datastream/connectors.py

Reply:
Good point with the visibility. I think it would be also good to add some doc strings for the static methods to symbol they are supposed to be used by users.

Reply:
> One thing left is that, pyflink has Pulsar connector support, but I don't know if that file is auto-generated or maintained by developers.
> 
> flink-python/pyflink/datastream/connectors.py

Unfortunately, it is not auto generated and manually maintained but I cannot really help with the python side.

Reply:
> > One thing left is that, pyflink has Pulsar connector support, but I don't know if that file is auto-generated or maintained by developers.
> > flink-python/pyflink/datastream/connectors.py
> 
> Unfortunately, it is not auto generated and manually maintained but I cannot really help with the python side.

I've added the java doc.  For python connector, I'll leave it intact and see what's our plan of updating the python connector. I think it' likely we can do a major upgrade after small fixes accumulates to a certain amount.